app/scripts/controllers/preferences.js

```diff
@@ -262,7 +254,7 @@ export default class PreferencesController {
     }, {});
 
     ///: BEGIN:ONLY_INCLUDE_IN(build-mmi)
-    this.prepareMmiPortfolio();
+    this.emit('update-mmi-portfolio');
```

Comment:
Note that the stores built into each controller already have state update events. You can use the pre-existing state update event instead of this new custom one.

e.g.

```
preferencesController.store.subscribe(() => {
  this.prepareMmiPortfolio(),
})
```

Or if you want to ensure it only runs when a certain piece of state changes, you could do something like

```
preferencesController.store.subscribe(
  previousValueComparator((prevState, currState) => {
    if (!isEqual(prevState.identities, currState.identities)) {
      this.prepareMmiPortfolio();
    }
  }, preferencesController.store.getState()),
)
```